lib/rules/prefer-object-spread.js

```diff
@@ -0,0 +1,259 @@
+    }
+
+    if (isLast) {
+        return parens ? `...(${text})` : `...${text}`;
+    }
+
+    return parens ? addComma(`...(${text})`) : `...${addComma(text)}`;
+}
+
+/**
+ * Autofixes the Object.assign call to use an object spread instead.
+ * @param {ASTNode|null} node - The node that the rule warns on, i.e. the Object.assign call
+ * @param {string} sourceCode - sourceCode of the Object.assign call
+ * @returns {Function} autofixer - replaces the Object.assign with a spread object.
+ */
+function autofixSpread(node, sourceCode) {
+    return fixer => {
+        const args = node.arguments;
+        const firstArg = args[0];
+        const lastArg = args[args.length - 1];
+        const parens = needsParens(node);
+        const replaceObjectAssignStart = fixer.replaceTextRange(
+            [node.range[0], firstArg.range[0]],
+            `${parens ? "({" : "{"}`
+        );
+
+        const handleArgs = args
+            .map((arg, i, arr) => formatArg(arg, i + 1 >= arr.length, sourceCode))
+            .filter(arg => arg !== "," && arg !== "");
+
+        const insertBody = fixer.replaceTextRange([firstArg.range[0], lastArg.range[1]], handleArgs.join(""));
+        const replaceObjectAssignEnd = fixer.replaceTextRange([lastArg.range[1], node.range[1]], `${parens ? "})" : "}"}`);
+
+        return [
+            replaceObjectAssignStart,
+            insertBody,
+            replaceObjectAssignEnd
+        ];
+    };
+}
+
+/**
+ * Autofixes the Object.assign call with a single object literal as an argument
+ * @param {ASTNode|null} node - The node that the rule warns on, i.e. the Object.assign call
+ * @param {string} sourceCode - sourceCode of the Object.assign call
+ * @returns {Function} autofixer - replaces the Object.assign with a object literal.
+ */
+function autofixObjectLiteral(node, sourceCode) {
+    return fixer => {
+        const argument = node.arguments[0];
+        const parens = needsParens(node);
+
+        return fixer.replaceText(node, `${parens ? "(" : ""}${sourceCode.text.slice(argument.range[0], argument.range[1])}${parens ? ")" : ""}`);
+    };
+}
+
+
+module.exports = {
+    meta: {
+        docs: {
+            description:
+                "disallow using Object.assign with an object literal as the first argument and prefer the use of object spread instead.",
+            category: "Stylistic Issues",
+            recommended: false,
+            url: "https://eslint.org/docs/rules/prefer-object-spread"
+        },
+        schema: [],
+        fixable: "code",
+        messages: {
+            useSpreadMessage: "Use an object spread instead of `Object.assign()` eg: `{ ...foo }`",
+            useLiteralMessage: "Use an object literal instead of `Object.assign`."
+        }
+    },
+
+    create: function rule(context) {
+        return {
+            CallExpression(node) {
+                const sourceCode = context.getSourceCode();
```

Comment:
Nit: You could move this a few lines up so that we don't need to retrieve the `SourceCode` object on every CallExpression. (It's a persistent reference.)